### DISCUSSION TITLE
Why make ConfigLoader.load as async ?

### DISCUSSION THREAD

**Discussion by shenshouer** [kube-rs/kube#1645] [Q&A]
I noticed that the load method in the ConfigLoader struct does not use await, so why is the load method marked as async?
https://github.com/kube-rs/kube/blob/main/kube-client/src/config/file_loader.rs#L56

**Comment by shenshouer** [comment]
I am very confused about this. Can someone answer this question?

**Comment by clux** [comment]
possibly an oversight the way `load` and `read` is split. certainly the loading is not doing any io at the moment at least, so the loading could be changed. howevere there's another discussion in  https://github.com/kube-rs/kube/discussions/1336 on whether we should make the file reading async, since that's also erroneously not marked async (and uses sync apis - hiding the fact that it is doing io and blocking), but we forgot about it. i'll raise an issue in a bit to consolidate this.

EDIT: issue in Issue now created in https://github.com/kube-rs/kube/issues/1651

**Comment by shenshouer** [comment]
I have submitted a PR https://github.com/kube-rs/kube/pull/1650 to change the load method to sync. Please help check if it meets the requirements.
